transport/controller.cc

```diff
@@ -112,47 +114,61 @@ future<> controller::do_start_server() {
         std::vector<listen_cfg> configs;
         int native_port_idx = -1, native_shard_aware_port_idx = -1;
 
-        if (cfg.native_transport_port.is_set() ||
```

Comment:
@margdoc This will be much easier to review if you split this commit as follows:
- the first commit adds the `if (!_maintenance_port)` but does not change indentation of code inside the `if` (from what I see, you're mostly putting a branch around pre-existing code)
- the second commit fixes the indentation inside the branch

Reply:
BTW:
"If a method has a boolean parameter and its content is enclosed in a big if on that parameter, it should really be two methods"
Julius Caesar, 40 BC